rust/processor/src/db/common/models/emojicoin_models/queries/insertion_queries.rs

```diff
@@ -233,6 +227,138 @@ pub fn insert_arena_melee_events_query(
+            .set((
+                rewards_remaining.eq(rewards_remaining + excluded(rewards_remaining)),
+                emojicoin_0_market_address.eq(excluded(emojicoin_0_market_address)),
+                emojicoin_1_market_address.eq(excluded(emojicoin_1_market_address)),
+                emojicoin_0_symbols.eq(excluded(emojicoin_0_symbols)),
+                emojicoin_1_symbols.eq(excluded(emojicoin_1_symbols)),
+                emojicoin_0_market_id.eq(excluded(emojicoin_0_market_id)),
+                emojicoin_1_market_id.eq(excluded(emojicoin_1_market_id)),
+                start_time.eq(excluded(start_time)),
+                duration.eq(excluded(duration)),
+                max_match_percentage.eq(excluded(max_match_percentage)),
+                max_match_amount.eq(excluded(max_match_amount)),
+            )),
+        None,
+    )
+}
+
+pub fn update_arena_info_query(
+    items_to_update: Vec<ArenaInfoDiffUpdate>,
+) -> (
+    impl QueryFragment<Pg> + diesel::query_builder::QueryId + Send,
+    Option<&'static str>,
+) {
+    use schema::arena_info::dsl::*;
+    let i: Vec<_> = items_to_update
+        .into_iter()
+        .map(|i| {
+            (
+                melee_id.eq(i.melee_id),
+                volume.eq(i.volume.clone()),
+                rewards_remaining.eq(i.rewards_remaining),
+                emojicoin_0_locked.eq(i.emojicoin_0_locked),
+                emojicoin_1_locked.eq(i.emojicoin_1_locked),
+            )
+        })
+        .collect();
+    (
+        diesel::insert_into(schema::arena_info::table)
+            .values(i)
+            .on_conflict(melee_id)
+            .do_update()
+            .set((
+                volume.eq(volume + excluded(volume)),
+                rewards_remaining.eq(rewards_remaining + excluded(rewards_remaining)),
+                emojicoin_0_locked.eq(emojicoin_0_locked + excluded(emojicoin_0_locked)),
+                emojicoin_1_locked.eq(emojicoin_1_locked + excluded(emojicoin_1_locked)),
+            )),
+        None,
+    )
+}
+
+pub fn insert_arena_leaderboard_history_query(
+    arena_leaderboard_history_model: ArenaLeaderboardHistoryPartialModel,
+) -> impl QueryFragment<Pg> + diesel::query_builder::QueryId + Send {
+    let mut query = include_str!("./leaderboard.sql").to_string();
+
+    // See header comment of leaderboard.sql for more information on query parameters.
+
+    // $1 is the melee_id
+    query = query.replace("$1", &arena_leaderboard_history_model.melee_id.to_string());
+
+    // $2 is the curve price of emojicoin_0
+    query = query.replace(
+        "$2",
+        &arena_leaderboard_history_model
+            .emojicoin_0_price
+            .to_string(),
+    );
+
+    // $3 is the curve price of emojicoin_1
+    query = query.replace(
+        "$3",
+        &arena_leaderboard_history_model
+            .emojicoin_1_price
+            .to_string(),
+    );
+
+    sql_query(query)
+}
+
+pub fn update_arena_leaderboard_history_query(
```

Comment:
```suggestion
pub fn update_user_last_exit_status_after_melee_has_ended_query(
```

Or a comment down below...or something just much more explicit- it's pretty vague like this and confusing when called in the code below